app/controllers/api/v2/chapters_controller.rb

```diff
@@ -3,57 +3,71 @@
 module Api
   module V2
     class ChaptersController < ApiController
-      before_action :find_chapter, only: %i[show changes headings]
```

Comment:
Removing this out of a callback because we might not need to fetch this from the database if we have something cached for the chapter id and actual date on the current service